internal/services/datafactory/data_factory_linked_service_sql_managed_instance_resource.go

```diff
@@ -0,0 +1,598 @@
+					ReferenceName: config.IntegrationRuntimeName,
+				}
+			}
+
+			if config.IntegrationRuntimeName != "" {
+				parameters.ConnectVia = &linkedservices.IntegrationRuntimeReference{
+					Type:          linkedservices.IntegrationRuntimeReferenceTypeIntegrationRuntimeReference,
+					ReferenceName: config.IntegrationRuntimeName,
+				}
+			}
+
+			if len(config.Annotations) > 0 {
+				annotations := make([]interface{}, len(config.Annotations))
+				for i, v := range config.Annotations {
+					annotations[i] = v
+				}
+				parameters.Annotations = &annotations
+			}
+
+			linkedService := linkedservices.LinkedServiceResource{
+				Properties: parameters,
+			}
+
+			if _, err := client.CreateOrUpdate(ctx, id, linkedService, linkedservices.DefaultCreateOrUpdateOperationOptions()); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r LinkedServiceSqlManagedInstanceResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.DataFactory.LinkedServicesClient
+
+			id, err := linkedservices.ParseLinkedServiceID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id, linkedservices.DefaultGetOperationOptions())
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", id, err)
+			}
+
+			if resp.Model == nil {
+				return fmt.Errorf("retrieving %s: `model` was nil", id)
+			}
+
+			if resp.Model.Properties == nil {
+				return fmt.Errorf("retrieving %s: `properties` was nil", id)
+			}
+
+			existing, ok := resp.Model.Properties.(linkedservices.AzureSqlMILinkedService)
+			if !ok {
+				return fmt.Errorf("classifying %s: Expected: %q Received: %T", id, "AzureSqlMI", resp.Model.Properties)
+			}
+
+			if existing.Type != "AzureSqlMI" {
+				return fmt.Errorf("classifying %s: Expected: %q Received: %q", id, "AzureSqlMI", existing.Type)
+			}
+
+			state := LinkedServiceSqlManagedInstanceModel{
+				Name:          id.LinkedServiceName,
+				DataFactoryID: factories.NewFactoryID(id.SubscriptionId, id.ResourceGroupName, id.FactoryName).ID(),
+			}
+
+			state.Description = pointer.From(existing.Description)
+
+			props := existing.TypeProperties
+
+			if props.ConnectionString != nil {
+				if val, ok := (*props.ConnectionString).(map[string]interface{}); ok {
+					state.KeyVaultConnectionString = flattenKeyVaultConnectionStringToConfig(val)
+				}
+				if val, ok := (*props.ConnectionString).(string); ok {
+					state.ConnectionString = val
+				}
+			}
+
+			if props.ServicePrincipalId != nil {
+				if id, ok := (*props.ServicePrincipalId).(string); ok {
+					state.ServicePrincipalID = id
+				}
+			}
```

Comment:
should we use `pointer.From` for these instead?

```suggestion
					state.ServicePrincipalID = pointer.From(props.ServicePrincipalId)
```